File: tests/fixtures/java/shadowing.java
package fixtures.shadowing;

class Shadowing {
    interface Callback {
        void run();
    }

    Callback make() {
        return null;
    }

    Callback target() {
        return null;
    }

    void shadowing() {
        Callback target = target();
        target.run();

        target = make();
        target.run();

        Callback first = make(), second = make();
        first.run();
        second.run();

        {
            Callback nested = make();
            nested.run();
        }
        nested.run();

        object.callback = make();
        object.run();
        values[0] = make();
        values.run();
    }
}

File: tests/fixtures/c/shadowing.c
typedef void (*Callback)(void);

Callback make(void);
void target(void) {}

void shadowing(void) {
    Callback target = ({
        target();
        make();
    });
    target();

    target = make();
    target();

    Callback first = make(), second = make();
    first();
    second();

    {
        Callback nested = make();
        nested();
    }
    nested();

    object.callback = make();
    object();
    values[0] = make();
    values();
}
